command/agent/local.go

```diff
@@ -177,7 +176,7 @@ func (l *localState) RemoveService(serviceID string) {
 
 	delete(l.services, serviceID)
 	delete(l.serviceTokens, serviceID)
-	l.serviceStatus[serviceID] = syncStatus{remoteDelete: true}
+	l.serviceStatus[serviceID] = syncStatus{inSync: false}
```

Comment:
Do we need to even set these? Since we are now just checking the `l.services` map directly, it seems like setting this flag won't really help.